tests/core/test_nlg.py

```diff
@@ -109,17 +109,58 @@ def test_nlg_schema_validation_empty_image():
 def test_nlg_fill_template_text(slot_name, slot_value):
     template = {"text": "{" + slot_name + "}"}
     t = TemplatedNaturalLanguageGenerator(templates=dict())
-    result = t._fill_template_text(
+    result = t._fill_template(
         template=template, filled_slots={slot_name: slot_value}
     )
     assert result == {"text": str(slot_value)}
 
 
+@pytest.mark.parametrize(
+    "slot_name, slot_value",
+    [
+        ("tag_w_underscore", "a"),
+        ("tag with space", "bacon"),
+        ("tag.with.dot", "chocolate"),
+        ("tag-w-dash", "apple pie"),
+        ("tag-w-$", "banana"),
+        ("tag-w-@", "one"),
+        ("tagCamelCase", "two"),
+        ("tag-w-*", "three"),
+        ("tag_w_underscore", "a"),
+        ("tag.with.float.val", 1.3),
+        ("tag-w-$", "banana"),
+        ("tagCamelCase", "two"),
```

Comment:
Can you maybe add some edge cases like `None` or empty string?

Reply:
@tabergma thoughts on how `None` should be treated? If we stringify it, it'll just be turned into `"None"` which isn't what we'd want?

Reply:
added a few edge cases to the tests